### ISSUE TITLE
MIUI 8 Notification Bar Issue 

### ISSUE THREAD

**Issue by iampawan** [marverenic/Jockey#111]
As you know MIUI 8 has white notification panel background. That is why when music player runs it shows white buttons in notification bar which is not visible .

**Comment by marverenic** [comment]
I don't keep up with third party ROMs, so I wasn't actually aware that MIUI implemented notifications differently than stock Android.

Are you using a beta version of Jockey 2.0? If so, then this might be a defect with the ROM itself. Jockey currently uses a `NotificationCompat.MediaStyle` notification style, and this behavior is entirely controlled by the system on Lollipop and higher (Quick research shows that MIUI 8 is based on Marshmallow).

By default, `NotificationCompat.MediaStyle` uses a dark background with white text. The background is customizable, but the text color is not. Just changing the button tint would affect the notifications appearance on all non-MIUI ROMs. I also don't think that there's a good way to change app behavior based on the ROM that's currently installed on the phone.

Unfortunately, I don't have a way of testing MIUI since I don't have a supported device and there does not appear to be an emulator image. If you're able to find a fix, please fork Jockey and open a PR.

**Comment by iampawan** [comment]
Your above comment is true and i totally understand it and aware of it. But
the problem is that other apps are also using NotificationCompat.
mediastyle but no such problem happening in them.
Well what i will ask you is that is it possible if we can make a custom
notification layout so that this bug can be fixed on every phone like some
otjer music players do.  It will also fix the album art bug which also
doesn't show in notification in some third party roms like miui and letv os

On 08-Aug-2016 9:47 pm, "Andrew Bailey" <notifications@github.com> wrote:

I don't keep up with third party ROMs, so I wasn't actually aware that MIUI
implemented notifications differently than stock Android.

Are you using a beta version of Jockey 2.0? If so, then this might be a
defect with the ROM itself. Jockey currently uses a
NotificationCompat.MediaStyle notification style, and this behavior is
entirely controlled by the system on Lollipop and higher (Quick research
shows that MIUI 8 is based on Marshmallow).

By default, NotificationCompat.MediaStyle uses a dark background with white
text. The background is customizable, but the text color is not. Just
changing the button tint would affect the notifications appearance on all
non-MIUI ROMs. I also don't think that there's a good way to change app
behavior based on the ROM that's currently installed on the phone.

Unfortunately, I don't have a way of testing MIUI since I don't have a
supported device and there does not appear to be an emulator image. If
you're able to find a fix, please fork Jockey and open a PR.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/marverenic/Jockey/issues/111#issuecomment-238288516>,
or mute the thread

**Comment by marverenic** [comment]
Making a custom notification layout isn't a great solution here. This is what Jockey used to do, and it's a bit fragile for a couple of reasons. 

The first reason is that this layout is different depending on the OS version. Versions of Jockey before 2.0 did this, and had four notification layouts (expanded and collapsed for pre-L and post-L). With the upcoming release of Nougat, this would have to increase to 6 because Google has changed the notification style. With `NotificationCompat.MediaStyle`, we don't need to make a single layout, and we always are using the latest notification style for free.

The other reason that this is a bad idea is that it doesn't really solve the problem. If other apps that use `NotificationCompat.MediaStyle` don't experience this behavior, then there's probably something wrong with Jockey's notification implementation.

If you don't mind, can you post a screenshot of what the expected and actual behavior is (i.e. a screenshot showing Google Play Music's notification and Jockey's notification side-by-side)?

**Comment by iampawan** [comment]
Yeah i will ask my friend and as soon as i will get it i will send you..
Any new update coming?
